sfdoc/publish/models.py

```diff
@@ -101,12 +101,7 @@ def get_absolute_url(self):
     def enqueue(self):
         self.status = self.STATUS_QUEUED
         self.time_queued = now()
-        self.error_message = ''
         self.save()
-        for article in self.articles.all():
-            article.delete()
-        for image in self.images.all():
-            image.delete()
```

Comment:
Sounds like there's a new invariant that a given bundle should only go through the queue once. Should we add a check to prevent attempts to enqueue a bundle that was already enqueued?

Reply:
No: its the the opposite. Now a bundle uploaded twice is treated identically to if it were two different bundles uploaded with the same content. All that matters is the content, not the bundle UUID.

Reply:
I understand that. But it now creates a new Bundle record in the Django db each time. So I'm saying maybe we want to make sure that it's not possible to enqueue the same Bundle record twice, since the intended use is creating a new Bundle record with the same UUID if you need to requeue.

Reply:
Okay I understand now. I could go either way on that.

enqueing is idempotent because there is no real queue datastructure (for the bundle queue).

So it would be harmless but might also indicate a bug in some new code. enqueue() is only called in two places and both of them create a brand new bundle just before so it seems a low risk both ways. Low risk of a bug being introduced and low risk of harm.

But it also takes only a minute to add

        assert self.status == self.STATUS_NEW

So I leave it up to you.

Reply:
I'll add the assertion.

Reply:
Agree that it wasn't super critical, but thanks for adding it.